### PR TITLE
Fix type of slice's rel_shape argument

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a bug in SliceAttr: ``rel_shape`` should be of type ``vector<float>``

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Fixed the type of ``rel_shape`` in SliceAttr*
 - Affected modules and functionalities:
     *Slice operator, when using the operator object API*
 - Key points relevant for the review:
     *NA*
 - Validation and testing:
     *NA*
 - Documentation (including examples):
     *NA*

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

**Review by awolant — APPROVED** [review]

File: dali/operators/generic/slice/slice_attr.cc [hunk 1]
```diff
@@ -61,7 +61,7 @@ is incompatible with providing positional inputs anchor and shape.)code",
 Providing named arguments ``start``, ``end``, ``shape``, ``rel_start``, ``rel_end``, ``rel_shape``
 is incompatible with providing positional inputs anchor and shape.)code",
         nullptr, true)
-    .AddOptionalArg<std::vector<int>>("rel_shape",
+    .AddOptionalArg<std::vector<float>>("rel_shape",
```
Comment (JanuszL) [inline dali/operators/generic/slice/slice_attr.cc:64]: Can you add a test for `rel_shape`?

Comment (jantonguirao) [reply dali/operators/generic/slice/slice_attr.cc:64]: There is a test, but this only throws an error when you use the object API (not the functional one). I wonder what to do about it.

**Review by JanuszL — APPROVED** [review]

File: dali/operators/generic/slice/slice_attr.cc (hunk 1 above)
Comment (jantonguirao) [reply dali/operators/generic/slice/slice_attr.cc:64]: I found that the test didn't check for python list arguments, so I extended (I verified the new tests failed before this fix)

**Comment by jantonguirao** [comment]
!build

**Review by mzient — APPROVED** [review]

**Comment by jantonguirao** [comment]
!build

**Review by JanuszL — APPROVED** [review]

**Pull request merged**